vendor/ansible/roles/swarm/templates/swarm.j2

```diff
@@ -8,12 +8,26 @@ fi
 
 case $1 in
 start)
-    echo starting swarm as on {{ node_name }}[{{ node_addr }}]
-    {% if swarm_bootstrap_node_name == node_name %}
-        /usr/bin/swarm manage -H tcp://{{ node_addr }}:{{ swarm_api_port }} etcd://{{ node_addr }}:{{ etcd_client_port1 }} &
+    echo starting swarm as {{ run_as }} on {{ node_name }}[{{ node_addr }}]
+    {% if run_as == "master" -%}
+    /usr/bin/docker run -t -d -p {{ swarm_api_port }}:{{ swarm_api_port }} --name=swarm-manager \
+        swarm:{{ swarm_version }} manage \
+        -H :{{ swarm_api_port }} \
+        --replication --advertise={{ node_addr }}:{{ swarm_api_port }} \
+        etcd://{{ node_addr }}:{{ etcd_client_port1 }}
     {% endif %}
+    /usr/bin/docker run -t --name=swarm-agent \
```

Comment:
> so, for the master nodes, we run both swarm manage and swarm join.

yes, that is by design swarm-manager is the daemon responsible for forming the swarm cluster so it is run on master.

> we might "conditionally" want to do swarm join only on the service-worker group

I think you meant `"conditionally" want to do swarm join only on the service-master`. If I understood it right, theoretically yes but practically it will always be fine to schedule containers on a swarm manager node as docker engine is guaranteed to be running there. Usually a host that is capable of running master would be able to participate as agent as well and we don't get that choice in some other clustered services like ucp for instance and other as well like etcd, consul etc, so it will be simple to try make these services fall just in master and worker groups.

Note that `host-capability` instead tracks physical capability of the host.